inference-engine/src/transformations/src/transformations/convert_opset4_to_opset3/convert_sequences_to_sequences_ie.cpp

```diff
@@ -0,0 +1,208 @@
+
+    ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
+        auto lstm_sequence = std::dynamic_pointer_cast<ngraph::opset4::LSTMSequence>(m.get_match_root());
+        if (!lstm_sequence) {
+            return false;
+        }
+
+        auto W = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
+                lstm_sequence->input_value(4).get_node_shared_ptr());
+        if (!W) {
+            return false;
+        }
+
+        auto R = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
+                lstm_sequence->input_value(5).get_node_shared_ptr());
+        if (!R) {
+            return false;
+        }
+
+        // for forward/reverse cases we can squeeze num_direction dimension
+        auto axis_1 = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1});
+        auto in_1 = std::make_shared<ngraph::opset4::Squeeze>(lstm_sequence->input(1).get_source_output(), axis_1);
+        auto in_2 = std::make_shared<ngraph::opset4::Squeeze>(lstm_sequence->input(2).get_source_output(), axis_1);
+        auto concat = std::make_shared<ngraph::opset4::Concat>(ngraph::NodeVector({W, R}), 2);
+        auto axis_2 = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {0});
+        auto in_3 = std::make_shared<ngraph::opset4::Squeeze>(concat->output(0), axis_2);
+        auto in_4 = std::make_shared<ngraph::opset4::Squeeze>(lstm_sequence->input(6).get_source_output(), axis_2);
+        auto lstm_sequence_ie = std::make_shared<ngraph::op::LSTMSequenceIE>(
+                lstm_sequence->input(0).get_source_output(),  // X
+                in_1,  // initial_hidden_state
+                in_2,  // initial_cell_state
+                //lstm_sequence->input(3).get_source_output(),  // sequence_lengths
+                in_3,                           // WR
+                in_4,  // B
+                lstm_sequence->get_hidden_size(),
+                lstm_sequence->get_direction(),
+                lstm_sequence->get_activations(),
+                lstm_sequence->get_activations_alpha(),
+                lstm_sequence->get_activations_beta(),
+                lstm_sequence->get_clip_threshold());
+
+        auto unsqueeze_axis = ngraph::opset4::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1});
+        auto unsqueeze_1 = std::make_shared<ngraph::opset4::Unsqueeze>(lstm_sequence_ie->output(0), unsqueeze_axis);
+        auto unsqueeze_2 = std::make_shared<ngraph::opset4::Unsqueeze>(lstm_sequence_ie->output(1), unsqueeze_axis);
+        auto unsqueeze_3 = std::make_shared<ngraph::opset4::Unsqueeze>(lstm_sequence_ie->output(2), unsqueeze_axis);
+        NodeVector unsqueezes = {unsqueeze_1, unsqueeze_2, unsqueeze_3};
+
+        for (int out_id = 0; out_id < lstm_sequence->outputs().size(); ++out_id) {
+            for (const auto& target_input : lstm_sequence->output(out_id).get_target_inputs()) {
```

Comment:
replace this cycle with 
```suggestion
lstm_sequence->output(out_id).replace(unsqueezes[out_id]);
```

Reply:
replaced on single replace_node() call